packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -0,0 +1,1277 @@
+        if (_dragItem != null) {
+          _dragUpdateItems();
+          _autoScroll();
+        }
+      }
+    }
+  }
+
+  SliverChildDelegate _createDelegate() {
+    return SliverChildBuilderDelegate(_itemBuilder, childCount: widget.itemCount + (_reorderingDrag ? 1 : 0));
+  }
+
+  Widget _itemBuilder(BuildContext context, int index) {
+    if (_dragInfo != null && index >= widget.itemCount) {
+      switch (_scrollDirection) {
+        case Axis.horizontal:
+          return SizedBox(width: _dragInfo!.itemExtent);
+        case Axis.vertical:
+          return SizedBox(height: _dragInfo!.itemExtent);
+      }
+    }
+    final Widget child = widget.itemBuilder(context, index);
+    assert(child.key != null, 'All list items must have a key');
+    final OverlayState overlay = Overlay.of(context)!;
+    return _ReorderableItem(
+      key: _ReorderableItemGlobalKey(child.key!, index, this),
+      index: index,
+      child: child,
+      capturedThemes: InheritedTheme.capture(from: context, to: overlay.context),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasOverlay(context));
+    return SliverList(
+      delegate: _createDelegate(),
+    );
+  }
+}
+
+class _ReorderableItem extends StatefulWidget {
+  const _ReorderableItem({
+    required Key key,
+    required this.index,
+    required this.child,
+    required this.capturedThemes,
+  }) : super(key: key);
+
+  final int index;
+  final Widget child;
+  final CapturedThemes capturedThemes;
+
+  @override
+  _ReorderableItemState createState() => _ReorderableItemState();
+}
+
+class _ReorderableItemState extends State<_ReorderableItem> {
+  late SliverReorderableListState _listState;
+
+  Offset _startOffset = Offset.zero;
+  Offset _targetOffset = Offset.zero;
+  AnimationController? _offsetAnimation;
+
+  Key get key => widget.key!;
+  int get index => widget.index;
+
+  bool get dragging => _dragging;
+  set dragging(bool dragging) {
+    if (mounted) {
+      setState(() {
+        _dragging = dragging;
+      });
+    }
+  }
+  bool _dragging = false;
+
+  @override
+  void initState() {
+    _listState = SliverReorderableList.of(context);
+    _listState._registerItem(this);
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _offsetAnimation?.dispose();
+    super.dispose();
+  }
+
+  @override
+  void didUpdateWidget(covariant _ReorderableItem oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.key != widget.key || oldWidget.index != widget.index) {
+      _listState._unregisterItem(oldWidget.index, this);
+      _listState._registerItem(this);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (_dragging) {
+      return const SizedBox();
+    }
+    return Transform(
+      transform: Matrix4.translationValues(offset.dx, offset.dy, 0.0),
+      child: widget.child,
+    );
+  }
+
+  @override
+  void deactivate() {
+    _listState._unregisterItem(index, this);
```

Comment:
What happens if the item is activated later? Isn't that possible when the item is moved within the list?

Reply:
Yeah, I should probably catch that in the build method.